buildSrc/src/main/kotlin/org/partiql/pig/gradle/publish/PigPublishPlugin.kt

```diff
@@ -0,0 +1,112 @@
+package org.partiql.pig.gradle.publish
+
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.plugins.JavaPlugin
+import org.gradle.api.plugins.JavaPluginExtension
+import org.gradle.api.publish.PublishingExtension
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.api.publish.maven.plugins.MavenPublishPlugin
+import org.gradle.api.publish.maven.tasks.PublishToMavenRepository
+import org.gradle.kotlin.dsl.create
+import org.gradle.kotlin.dsl.get
+import org.gradle.kotlin.dsl.getByName
+import org.gradle.kotlin.dsl.provideDelegate
+import org.gradle.plugins.signing.SigningExtension
+import org.gradle.plugins.signing.SigningPlugin
+import org.jetbrains.dokka.gradle.DokkaPlugin
+import org.jetbrains.dokka.gradle.DokkaTask
+import java.io.File
+
+/**
+ * Gradle plugin to consolidates the following publishing logic
+ * - Maven Publising
+ * - Signing
+ * - SourcesJar
+ * - Dokka + JavadocJar
+ */
+abstract class PigPublishPlugin : Plugin<Project> {
+
+    override fun apply(target: Project): Unit = with(target) {
+        pluginManager.apply(JavaPlugin::class.java)
+        pluginManager.apply(MavenPublishPlugin::class.java)
+        pluginManager.apply(SigningPlugin::class.java)
+        pluginManager.apply(DokkaPlugin::class.java)
+        val ext = extensions.create("publish", PublishExtension::class.java)
+        target.afterEvaluate { publish(ext) }
+    }
+
+    private fun Project.publish(ext: PublishExtension) {
+        val releaseVersion = !version.toString().endsWith("-SNAPSHOT")
+
+        // Generate "javadoc"
+        tasks.getByName<DokkaTask>("dokkaHtml") {
+            outputDirectory.set(File("${buildDir}/javadoc"))
+        }
+
+        // Include "sources" and "javadoc" in the JAR
+        extensions.getByType(JavaPluginExtension::class.java).run {
+            withSourcesJar()
+            withJavadocJar()
+        }
+
```

Comment:
Non-blocking -- do we need to add the dokkaHtml output to the javadocJar? Similar to how we do it in PLK?